tests/test_modeling_tf_encoder_decoder.py

```diff
@@ -240,7 +240,7 @@ def check_encoder_decoder_model_labels(
         assert "loss" in outputs_encoder_decoder
 
         batch_size, seq_len = decoder_input_ids.shape
-        expected_shape = (batch_size, seq_len - 1, decoder_config.vocab_size)
+        expected_shape = (batch_size, seq_len, decoder_config.vocab_size)
```

Comment:
This is changed, because in TF Causal LM models, the returned logits is the truncated one `logits = logits[:, :-1]`.
However, in PyTorch Causal LM models, the returned logits is 

```
        prediction_scores = self.cls(sequence_output)

        if labels is not None:
            # we are doing next-token prediction; shift prediction scores and input ids by one
            shifted_prediction_scores = prediction_scores[:, :-1, :].contiguous()
            ...
            lm_loss = loss_fct(shifted_prediction_scores.view(-1, self.config.vocab_size), labels.view(-1))

        return CausalLMOutputWithCrossAttentions(
            loss=lm_loss,
            logits=prediction_scores,
            ...
```
When using the same logic as in #14139 to compute loss independent from decoder, there is no truncation needed to be done.